[K8S] Documentation kubeconfig with IAM (#2575)
scaleway/docs-content: containers/kubernetes/how-to/manage-kubeconfig-with-iam.mdx

---
meta:
  title: How to manage Kubeconfig files with IAM
  description: This page explains how to manage Scaleway Kubeconfig files with IAM
content:
  h1: How to manage Kubeconfig files with IAM
  paragraph: This page explains how to manage Scaleway Kubeconfig files with IAM
tags: kubernetes kapsule-cluser
dates:

>   validation: 2024-07-30
>   posted: 2024-07-  

Author: ```suggestion
  validation: 2024-11-05
  posted: 2024-11-05
```

categories:
  - kubernetes
---

> 

Reviewer: ```suggestion
```

A Kubeconfig file serves as a repository for essential credentials required to access a Kubernetes cluster and manage deployments.  
This file contains critical information, including cluster details, authentication data, and context information, allowing users to oversee and configure their Kubernetes clusters.

Scaleway's Identity and Access Management (IAM) is now being used to generate Kubeconfig files. This integration offers several advantages:

- **Improved user administration:**
   - When removing a cluster user, there is no longer a need to reset the admin token, mitigating security risks.
- **Increased cluster security:**
   - Each user has unique credentials, enhancing security.
   - Audit logs facilitate the identification of specific Kubernetes users.
- **Enhanced application security:**
   - Applications within the cluster are authenticated using specific tokens.
   - This authentication method is particularly useful when a pod needs to interact with the Kubernetes API, as it can authenticate using its associated [Service Account](https://kubernetes.io/docs/concepts/security/service-accounts/).

> [Identity and Access Management (IAM)](https://www.scaleway.com/en/docs/identity-and-access-management/iam/concepts/#iam) provides control over resource access. IAM policies enable the configuration of permissions for Kubernetes Kapsule clusters at the Project level.

Reviewer: ```suggestion
[Identity and Access Management (IAM)](/identity-and-access-management/iam/concepts/#iam) provides control over resource access. IAM policies enable the configuration of permissions for Kubernetes Kapsule clusters at the Project level.
```

> An [IAM policy](https://www.scaleway.com/en/docs/identity-and-access-management/iam/concepts/#policy) defines the permissions for users, groups, and applications within an Organization. It consists of a [principal](https://www.scaleway.com/en/docs/identity-and-access-management/iam/concepts/#principal) (the user, group, or application to which it applies) and IAM rules that specify permission sets and their scope.

Reviewer: ```suggestion
An [IAM policy](/identity-and-access-management/iam/concepts/#policy) defines the permissions for users, groups, and applications within an Organization. It consists of a [principal](/identity-and-access-management/iam/concepts/#principal) (the user, group, or application to which it applies) and IAM rules that specify permission sets and their scope.
```

The [combination of IAM and Kubernetes RBAC (Role-based Acccess Control)](/containers/kubernetes/reference-content/set-iam-permissions-and-implement-rbac/) allows you to define fine-grained access levels for cluster users.

<Message type="important">

>     Entering a deprecation cycle, the former authentication system (using static admin tokens) remains active until the second half of 2024: all your legacy Kubeconfig files keep access to their clusters for the time being. For security purposes, you can still use the ["reset admin token" feature](/containers/kubernetes/how-to/revoke-kubeconfig/), either through our Kubernetes API or Scaleway’s CLI.

Reviewer: Is it really second half of 2024? Or Q1 2025, as we're mentioning on the console?

    We recommend that you download and use new unique, identifiable Kubeconfig files with IAM tokens inside.
</Message>

<Macro id="iam-requirements" />

<Message type="requirement">

>   - You have an account and are logged into the [Scaleway console](https://console.scaleway.com)

Reviewer: ```suggestion
  - A Scaleway account logged into the [console](https://console.scaleway.com/)
```

>   - You have [created](/containers/kubernetes/how-to/create-cluster) a Kubernetes Kapsule cluster

Reviewer: ```suggestion
  - [Created](/containers/kubernetes/how-to/create-cluster) a Kubernetes Kapsule cluster
```

---
meta:
  title: How to manage Kubeconfig files with IAM
  description: This page explains how to manage Kubeconfig files with IAM
content:
  h1: How to manage Kubeconfig files with IAM
  paragraph: This page explains how to manage Kubeconfig files with IAM
tags: kubernetes kapsule-cluser
dates:
  validation: 2024-01-09
  posted: 2024-01-09
categories:
  - kubernetes
---

A Kubeconfig file serves as a repository for essential credentials required to access a Kubernetes cluster and manage deployments.  
This file contains critical information, including cluster details, authentication data, and context information, allowing users to oversee and configure their Kubernetes clusters.

Scaleway's Identity and Access Management (IAM) is now being used to generate Kubeconfig files. This integration offers several advantages:

- **Improved user administration:**
   - When removing a cluster user, there is no longer a need to reset the admin token, mitigating security risks.

> - **Increased Cluster Security:**

Reviewer 2: ```suggestion
- **Increased cluster security:**
```

>    - Each user possesses a unique credential, enhancing security.

Reviewer 3: ```suggestion
   - Each user has unique credentials, enhancing security.
```

   - Audit logs facilitate the identification of specific Kubernetes users.

> - **Enhanced Application Security:**

Reviewer 2: ```suggestion
- **Enhanced application security:**
```

   - Applications within the cluster are authenticated using specific tokens.
   - This authentication method is particularly useful when a pod needs to interact with the Kubernetes API, as it can authenticate using its associated Service Account.

[Identity and Access Management (IAM)](https://www.scaleway.com/en/docs/identity-and-access-management/iam/concepts/#iam) provides control over resource access. IAM policies enable the configuration of permissions for Kubernetes Kapsule clusters at the Project level.

An [IAM policy](https://www.scaleway.com/en/docs/identity-and-access-management/iam/concepts/#policy) defines the permissions for users, groups, and applications within an Organization. It consists of a [principal](https://www.scaleway.com/en/docs/identity-and-access-management/iam/concepts/#principal) (the user, group, or application to which it applies) and IAM rules that specify permission sets and their scope.

> The [combination of IAM and Kubernetes RBAC](/containers/kubernetes/reference-content/set-iam-permissions-and-implement-rbac/) allows to define finely granulated access levels for cluster users.

Author: ```suggestion
The [combination of IAM and Kubernetes RBAC](/containers/kubernetes/reference-content/set-iam-permissions-and-implement-rbac/) allows defining fine-grained access levels for cluster users.
```

> The [combination of IAM and Kubernetes RBAC](http://localhost:8003/containers/kubernetes/reference-content/set-iam-permissions-and-implement-rbac/) allows to define finely granulated access levels for cluster users.

Reviewer: ```suggestion
The [combination of IAM and Kubernetes RBAC](http://localhost:8003/containers/kubernetes/reference-content/set-iam-permissions-and-implement-rbac/) allows defining fine-grained access levels for cluster users.
```

> The [combination of IAM and Kubernetes RBAC (Role-based Acccess Control)](/containers/kubernetes/reference-content/set-iam-permissions-and-implement-rbac/) allows defining fine-grained access levels for cluster users.

Reviewer 2: ```suggestion
The [combination of IAM and Kubernetes RBAC (Role-based Acccess Control)](/containers/kubernetes/reference-content/set-iam-permissions-and-implement-rbac/) allows you to define fine-grained access levels for cluster users.
```

<Message type="important">

>     Entering a deprecation cycle, the former authentication system (using static admin tokens) remains active until Q4/2024: all your legacy Kubeconfig files keep access to their clusters for the time being. For security measure you can still make use of the "reset admin token" feature !!!ADD LINK HERE!!!, either through our Kubernetes API or Scaleway’s CLI.

Reviewer: ```suggestion
    Entering a deprecation cycle, the former authentication system (using static admin tokens) remains active until Q4/2024: all your legacy Kubeconfig files keep access to their clusters for the time being. For security measure, you can still use the "reset admin token" feature !!!ADD LINK HERE!!!, either through our Kubernetes API or Scaleway’s CLI.
```

>     Entering a deprecation cycle, the former authentication system (using static admin tokens) remains active until Q4/2024: all your legacy Kubeconfig files keep access to their clusters for the time being. For security measure you can still make use of the "reset admin token" feature !!!ADD LINK HERE!!!, either through our Kubernetes API or Scaleway’s CLI.

Reviewer 3: Just a reminder to add the !!!LINK HERE!!! hehe

Author: Thanks. Added the link :)

>     Entering a deprecation cycle, the former authentication system (using static admin tokens) remains active until Q4/2024: all your legacy Kubeconfig files keep access to their clusters for the time being. For security measure, you can still use the ["reset admin token" feature](/containers/kubernetes/how-to/revoke-kubeconfig/), either through our Kubernetes API or Scaleway’s CLI.

Reviewer 4: ```suggestion
    Entering a deprecation cycle, the former authentication system (using static admin tokens) remains active until the second half of 2024: all your legacy Kubeconfig files keep access to their clusters for the time being. For security purposes, you can still use the ["reset admin token" feature](/containers/kubernetes/how-to/revoke-kubeconfig/), either through our Kubernetes API or Scaleway’s CLI.
```

>     We recommend users to download new unique, identifiable Kubeconfig files with IAM tokens inside.

Author: ```suggestion
    We recommend to download and use new unique, identifiable Kubeconfig files with IAM tokens inside.
```

>     We recommend to download and use new unique, identifiable Kubeconfig files with IAM tokens inside.

Reviewer 2: ```suggestion
    We recommend that you download and use new unique, identifiable Kubeconfig files with IAM tokens inside.
```

</Message>

<Macro id="iam-requirements" />

<Message type="requirement">
  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com)
  - You have [created](/containers/kubernetes/how-to/create-cluster) a Kubernetes Kapsule cluster
</Message>

## Setting up access

> 1. [Create a group](/identity-and-access-management/iam/how-to/create-group/) named `k8s-group` that contains the user and the application, so they are the only ones to have access to Object Storage.

Author: ```suggestion
1. [Create a group](/identity-and-access-management/iam/how-to/create-group/) named `k8s-group` that contains the user and the application, so they are the only ones to have access to your cluster.
```

> 2. [Create an IAM policy](/identity-and-access-management/iam/how-to/create-policy/) named `k8s-policy` in the Scaleway console, with the previously created group as a [principal](/identity-and-access-management/iam/concepts/#principal). Apply the corresponding IAM policy to the group. For example `KubernetesFullAccess`.

Reviewer: ```suggestion
2. [Create an IAM policy](/identity-and-access-management/iam/how-to/create-policy/) named `k8s-policy` in the Scaleway console, with the previously created group as a [principal](/identity-and-access-management/iam/concepts/#principal). Apply the corresponding IAM policy to the group. For example, `KubernetesFullAccess`.
```
